reactor-core/src/test/java/reactor/core/publisher/FluxHandleTest.java

```diff
@@ -510,4 +512,102 @@ public void contextTestFilterHide() {
+		StepVerifier.create(Flux.just(1)
+		                        .hide()
+		                        .filter(i -> true)
+		                        .handle((v, sink) -> {
+			                        sink.complete();
+			                        sink.next(2);
+		                        }))
+		            .verifyErrorSatisfies(e -> assertThat(e).isInstanceOf(IllegalStateException.class)
+		            .hasMessage("Cannot emit after a complete or error"));
+	}
+
+	@Test
+	public void nextAfterErrorNormalConditional() {
+		StepVerifier.create(Flux.just(1)
+		                        .hide()
+		                        .filter(i -> true)
+		                        .handle((v, sink) -> {
+			                        sink.error(new NullPointerException("boom"));
+			                        sink.next(2);
+		                        }))
+		            .verifyErrorSatisfies(e -> assertThat(e).isInstanceOf(IllegalStateException.class)
+		                                                    .hasMessage("Cannot emit after a complete or error"));
+	}
+
+	@Test
+	public void nextAfterCompleteFused() {
+		StepVerifier.create(Flux.just(1)
+		                        .handle((v, sink) -> {
+			                        sink.complete();
+			                        sink.next(2);
+		                        }))
+		            .verifyErrorSatisfies(e -> assertThat(e).isInstanceOf(IllegalStateException.class)
+		            .hasMessage("Cannot emit after a complete or error"));
+	}
+
+	@Test
+	public void nextAfterErrorFused() {
+		StepVerifier.create(Flux.just(1)
+		                        .handle((v, sink) -> {
+			                        sink.error(new NullPointerException("boom"));
+			                        sink.next(2);
+		                        }))
+		            .verifyErrorSatisfies(e -> assertThat(e).isInstanceOf(IllegalStateException.class)
+		                                                    .hasMessage("Cannot emit after a complete or error"));
+	}
+
+	@Test
+	public void nextAfterCompleteFusedConditional() {
+		StepVerifier.create(Flux.just(1)
+		                        .filter(i -> true)
+		                        .handle((v, sink) -> {
+			                        sink.complete();
+			                        sink.next(2);
+		                        }))
+		            .verifyErrorSatisfies(e -> assertThat(e).isInstanceOf(IllegalStateException.class)
+		            .hasMessage("Cannot emit after a complete or error"));
+	}
+
+	@Test
+	public void nextAfterErrorFusedConditional() {
+		StepVerifier.create(Flux.just(1)
+		                        .filter(i -> true)
+		                        .handle((v, sink) -> {
+			                        sink.error(new NullPointerException("boom"));
+			                        sink.next(2);
+		                        }))
+		            .verifyErrorSatisfies(e -> assertThat(e).isInstanceOf(IllegalStateException.class)
+		                                                    .hasMessage("Cannot emit after a complete or error"));
+	}
+
```

Comment:
Be good to add the other cases while here:
- error() + error()
- next() + next()
- complete() + error()